it/main/PostgresTest.java

```diff
@@ -106,21 +106,42 @@ public DimensionalDataSet createEmptyDataset(EntityManager em, String id, String
         return dimensionalDataSet;
     }
 
-    public void loadEachLineInV3File(EntityManager em, String inputFileName, DimensionalDataSet dimensionalDataSet) throws IOException {
+    public void loadEachLineInV3File(EntityManager em, String inputFileName, DimensionalDataSet dimensionalDataSet) throws IOException, DatapointMappingException {
         String rowData[];
         InputCSVParserV3 parser = new InputCSVParserV3();
-        BufferedReader csvReader = parser.getCSVBufferedReader(new File(new PostgresTest().getClass().getResource(inputFileName).getPath()));
+        BufferedReader csvReader = getCSVBufferedReader(new File(new PostgresTest().getClass().getResource(inputFileName).getPath()));
```

Comment:
This can all be somewhat simplified:

```java
        try (BufferedReader csvReader = new BufferedReader(new InputStreamReader(getClass().getResourceAsStream(inputFileName), "UTF-8"), 32768)) {
            
        }
```
Then you can get rid of the `getCSVBufferedReader` and `closeCSVReader` methods.